src/name.rs

```diff
@@ -539,10 +539,17 @@ impl NamespaceResolver {
             .unwrap_or_else(|| Self::maybe_unknown(prefix))
     }
 
+    const XML_NAMESPACE: &'static [u8] = b"http://www.w3.org/XML/1998/namespace";
+    const XMLNS_NAMESPACE: &'static [u8] = b"http://www.w3.org/2000/xmlns/";
+
     #[inline]
     fn maybe_unknown(prefix: Option<Prefix>) -> ResolveResult<'static> {
         match prefix {
-            Some(p) => ResolveResult::Unknown(p.into_inner().to_vec()),
+            Some(p) => match p.into_inner() {
+                b"xml" => ResolveResult::Bound(Namespace(NamespaceResolver::XML_NAMESPACE)),
+                b"xmlns" => ResolveResult::Bound(Namespace(NamespaceResolver::XMLNS_NAMESPACE)),
+                x => ResolveResult::Unknown(x.to_vec()),
+            },
```

Comment:
This looks better:
```suggestion
            Some(Prefix(b"xml")) => ResolveResult::Bound(Self::XML_NAMESPACE),
            Some(Prefix(b"xmlns")) => ResolveResult::Bound(Self::XMLNS_NAMESPACE),
            Some(p) => ResolveResult::Unknown(p.into_inner().to_vec()),
```
Also, need to add a TODO comment, that this will not work with UTF-16 and ISO_2022_JP encodings, because namespaces stored in document encoding. As written it will work only because all other [supported encodings](https://docs.rs/encoding_rs/0.8.31/encoding_rs/index.html#no-extensibility-by-design) are ASCII-compatible.

Reply:
I have pulled the definition of the prefixes and the actual mapping to be next to one another in the code. I have used the Prefixes in a more abstract manner that I think obviates the needs to TODO this code since changing the implementation of Prefix is what appears to be needed to fix the non-ascii compat encodings. Please let me know if you agree.